futures/tests/test_macro.rs

```diff
@@ -0,0 +1,18 @@
+#[cfg(test)]
+mod tests {
+    #[futures_test::test]
+    async fn it_works() {
+        let fut = async { true };
+        assert!(fut.await);
+
+        let fut = async { false };
+        assert!(!fut.await);
+    }
+
+    #[futures_test::test]
+    #[should_panic]
+    async fn it_is_being_run() {
```

Comment:
I don't know if there is a reliable way to test if the generate test actually gets run (has the `test` attribute).

Reply:
This is good as-is for now.
[macrotest](https://crates.io/crates/macrotest) is a crate for that use case ([example](https://github.com/taiki-e/pin-project/tree/main/tests/expand)), but it's difficult to use with `#[futures_test::test]`, as it will show other code generated by `#[test]`.